### PR TITLE
Simplify gopher.js TCP sockets sample

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

File: samples/tcp/gopher.js [hunk 1]
```diff
@@ -32,19 +29,11 @@ export default {
...
-        if (res.done) {
-          console.log("Stream done, socket connection has been closed.");
-          // TODO(soon): Half-open connections:
-          // see https://github.com/cloudflare/workerd/pull/162#discussion_r1020483277.
-          break;
-        }
-        gopherResponse += decoder.decode(res.value);
+      for await (const chunk of socket.readable) {
+        gopherResponse += decoder.decode(chunk);
       }
```
Comment (jasnell) [inline samples/tcp/gopher.js:37]: For decoding, you'll want...

```suggestion
      for await (const chunk of socket.readable) {
        gopherResponse += decoder.decode(chunk, { stream: true });
      }
      gopherResponse += decoder.decode(chunk);
```

This ensures that if the chunk cuts off any multi-byte characters they'll be properly handled.

Comment (dom96) [reply samples/tcp/gopher.js:37]: I assume that last `gopherResponse += decoder.decode(chunk);` line is a mistake

Comment (jasnell) [reply samples/tcp/gopher.js:37]: Doh, yes, it should be `gopherResponse += decoder.decode()` with no chunk. It's necessary to flush any remaining items from the decoders buffer.

Comment (jasnell) [reply samples/tcp/gopher.js:37]:
```suggestion
      for await (const chunk of socket.readable) {
        gopherResponse += decoder.decode(chunk, { stream: true });
      }
      gopherResponse += decoder.decode();
```

Comment (dom96) [reply samples/tcp/gopher.js:37]: or is it? but then how does this work? I would expect the `chunk` to be only accessible inside the for loop.

Comment (dom96) [reply samples/tcp/gopher.js]: Surprisingly it worked. Does the `chunk` exist outside the `for` loop scope?

**Review by jasnell — APPROVED** [review]

**Pull request merged**